### ISSUE TITLE
Failure to fetch data / requests.get returns 503 while curl is 200 

### ISSUE THREAD

**Issue by Hu1buerger** [psf/requests#6394]
requests.get fails to fetch page, while curl on the same machine gets a 200.
Additionaly the same code runs on macos 13.0.1

## Expected Result

```py
import requests # with version 2.28.2

res = requests.get("https://www.reichelt.com/raspberry-pi-zero-2-w-4x-1-ghz-512-mb-ram-wlan-bt-rasp-pi-zero2-w-p313902.html?CCOUNTRY=445&LANGUAGE=de")
assert res.status_code == 200
```

with `curl "https://www.reichelt.com/raspberry-pi-zero-2-w-4x-1-ghz-512-mb-ram-wlan-bt-rasp-pi-zero2-w-p313902.html?CCOUNTRY=445&LANGUAGE=de"` which returns the expected page.

Both ran on the same machine with python 3.10.

## Actual Result

`requests.get` returns http `503`

## Reproduction Steps

see above.

## System Information

    $ python -m requests.help

```json
{
  "chardet": {
    "version": "4.0.0"
  },
  "charset_normalizer": {
    "version": "3.1.0"
  },
  "cryptography": {
    "version": "3.4.8"
  },
  "idna": {
    "version": "3.4"
  },
  "implementation": {
    "name": "CPython",
    "version": "3.10.6"
  },
  "platform": {
    "release": "5.15.0-1025-raspi",
    "system": "Linux"
  },
  "pyOpenSSL": {
    "openssl_version": "30000020",
    "version": "21.0.0"
  },
  "requests": {
    "version": "2.28.2"
  },
  "system_ssl": {
    "version": "30000020"
  },
  "urllib3": {
    "version": "1.26.15"
  },
  "using_charset_normalizer": false,
  "using_pyopenssl": true
}

```

Pi 4 
with Ubuntu 22.04.2 LTS

**Comment by nateprewitt** [comment]
Hi @Hu1buerger, a 503 is a server side error. There's nothing we can do about this in Requests. You'll need to determine what's failing on the server in this case and whether it's intentional to block web scraping.

I'm going to resolve this as we can't take any action with the current info.

**Comment by hu1buerger** [comment]
@nateprewitt thats what i am refering to. I am assuming this is a clientside issue, as curling from the server is working fine. also running the same code on macos is working.

Why i believe this to be the case:

The server gets the same request headers (i.e. UserAgent), and assumingly the same content from the same source ip.
There should not be any destinguishing features between either request (send by requests on mac, requests on linux and curl on linux). All requests are send roughly at the same time (ie. rate limiting, spam protection)

**Comment by nateprewitt** [comment]
> I am assuming this is a clientside issue

Yes, it may be due to a client side behavior, but we cannot debug all server implementations on the web.

If there is a root cause for the server side failure and clear indication of a defect in Requests, we can potentially investigate that.

**Comment by hu1buerger** [comment]
@nateprewitt imo the following test should be satisfied. especially as it works on macos but not on linux

```py
import subprocess
import unittest

import requests

REICHELT_URL = "https://www.reichelt.com/raspberry-pi-zero-2-w-4x-1-ghz-512-mb-ram-wlan-bt-rasp-pi-zero2-w-p313902.html?CCOUNTRY=445&LANGUAGE=de"


class MyTestCase(unittest.TestCase):

    def test_requests(self):
        res = requests.get(REICHELT_URL)
        self.assertEqual(res.status_code, 200)

    def test_curl_should_be_200(self):
        cmd = "curl -s -o /dev/null -w \"%{http_code}\"" + f" \"{REICHELT_URL}\""
        res = subprocess.check_output(cmd, shell=True)
        http_status = int(res)
        self.assertEqual(http_status, 200)


if __name__ == '__main__':
    unittest.main()

```

**Comment by hu1buerger** [comment]
TLDR; it was a security check; see ` 'Security Check' in res.text`
